File: WitchcraftApp/src/main/java/com/tarbadev/witchcraft/weeks/persistence/entity/WeekEntity.java
package com.tarbadev.witchcraft.weeks.persistence.entity;

import com.tarbadev.witchcraft.weeks.domain.entity.Week;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.util.List;
import java.util.stream.Collectors;

import static java.util.stream.Collectors.toList;

@Builder
@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity(name = "weeks")
public class WeekEntity {
  @Id
  @GeneratedValue(strategy= GenerationType.AUTO)
  private Integer id;
  private Integer year;
  private Integer weekNumber;
  @OneToMany(cascade = CascadeType.ALL)
  @JoinColumn(name = "week_id")
  private List<DayEntity> days;

  public WeekEntity(Week week) {
    id = week.getId();
        year = week.getYear();
    weekNumber = week.getWeekNumber();
        days = week.getDays().stream().map(DayEntity::new).collect(toList());
  }

  public Week week() {
    return Week.builder()
        .id(id)
        .weekNumber(weekNumber)
        .year(year)
        .days(days.stream().map(DayEntity::day).collect(toList()))
        .build();
  }
}

File: WitchcraftApp/src/main/kotlin/com/tarbadev/witchcraft/weeks/persistence/entity/WeekEntity.kt
package com.tarbadev.witchcraft.weeks.persistence.entity

import com.tarbadev.witchcraft.weeks.domain.entity.Week
import javax.persistence.*

@Entity(name = "week")
data class WeekEntity(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private val id: Int = 0,
    private val year: Int = 0,
    private val weekNumber: Int = 0,
    @OneToMany(cascade = [CascadeType.ALL])
    @JoinColumn(name = "week_id")
    private val days: List<DayEntity> = emptyList()
) {
    constructor(week: Week) : this(
        id = week.id,
        year = week.year,
        weekNumber = week.weekNumber,
        days = week.days.map { DayEntity(it) }
    )

    fun week(): Week {
        return Week(
            id = id,
            weekNumber = weekNumber,
            year = year,
            days = days.map { it.day() }.sortedBy { it.name }
        )
    }
}
